### PR TITLE
feat(ext/crypto): export RSA as pkcs#8

### PR DESCRIPTION
**Pull request by littledivy**
towards #11690

### CODE REVIEW

File: ext/crypto/lib.rs [hunk 1]
```diff
@@ -522,6 +523,96 @@ pub async fn op_crypto_verify_key(
...
+          let pk_info = rsa::pkcs8::PrivateKeyInfo {
+            attributes: None,
+            public_key: None,
+            algorithm: rsa::pkcs8::AlgorithmIdentifier {
+              // Spec wants the OID to be id-RSASSA-PSS (1.2.840.113549.1.1.10) but ring and RSA do not support it.
+              // Instead, we use rsaEncryption (1.2.840.113549.1.1.1) as specified in RFC 3447.
+              // Node, Chromium and Firefox also use rsaEncryption (1.2.840.113549.1.1.1) and do not support id-RSASSA-PSS.
+
+              // parameters are set to NULL opposed to what spec wants (see above)
+              oid: rsa::pkcs8::ObjectIdentifier::new("1.2.840.113549.1.1.10"),
```
Comment (littledivy) [inline ext/crypto/lib.rs:599]: This makes us go off-spec but there is no other choice. Blink and Gecko also does not support  `id-RSASSA-PSS` https://www.chromium.org/blink/webcrypto#TOC-Supported-key-formats

Also, [See this note about interoperability](https://w3c.github.io/webcrypto/#SubtleCrypto-method-importKey).

Comment (littledivy) [reply ext/crypto/lib.rs]: EDIT: won't really call off-spec :-)

File: ext/crypto/lib.rs [hunk 2]
```diff
@@ -552,6 +554,137 @@ pub async fn op_crypto_verify_key(
...
+          let pk_info = rsa::pkcs8::PrivateKeyInfo {
+            attributes: None,
+            public_key: None,
+            algorithm: rsa::pkcs8::AlgorithmIdentifier {
+              // Spec wants the OID to be id-RSASSA-PSS (1.2.840.113549.1.1.10) but ring and RSA do not support it.
+              // Instead, we use rsaEncryption (1.2.840.113549.1.1.1) as specified in RFC 3447.
+              // Node, Chromium and Firefox also use rsaEncryption (1.2.840.113549.1.1.1) and do not support id-RSASSA-PSS.
+
+              // parameters are set to NULL opposed to what spec wants (see above)
+              oid: rsa::pkcs8::ObjectIdentifier::new("1.2.840.113549.1.1.1"),
```
Comment (ry) [inline ext/crypto/lib.rs:632]: So probably we shouldn't support it either...

**Review by lucacasonato — COMMENTED** [review]
Are there no WPT for this?

**Comment by littledivy** [comment]
> Are there no WPT for this?

@lucacasonato There are. Will enable them in #11891 :-)

File: ext/crypto/lib.rs (hunk 2 above)
Comment (littledivy) [reply ext/crypto/lib.rs:632]: yup, that is the current behavior :+1:

**Review by lucacasonato — APPROVED** [review]
LGTM! Thanks :-)

**Pull request merged**